### PR TITLE
First Pull Request

### PR DESCRIPTION
**Pull request by nicolbidone**
evidence

### CODE REVIEW

File: app/src/main/java/com/example/nicocalculator/MainActivity.kt [hunk 1]
```diff
@@ -2,11 +2,20 @@ package com.example.nicocalculator
...
 import android.os.Bundle
+import kotlinx.android.synthetic.main.activity_main.*
 
 class MainActivity : AppCompatActivity() {
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
+
+        text_results.text = "Result"
```
Comment (SartiMau) [inline app/src/main/java/com/example/nicocalculator/MainActivity.kt]: No hardcode please. Use resources or constants

File: app/src/main/java/com/example/nicocalculator/MainActivity.kt [hunk 2]
```diff
@@ -2,11 +2,20 @@ package com.example.nicocalculator
...
+        text_operations.text = "Operations"
+
+        button_0.setOnClickListener {
+            text_results.text = "pressed"
```
Comment (SartiMau) [inline app/src/main/java/com/example/nicocalculator/MainActivity.kt]: same

File: app/src/main/res/layout/activity_main.xml [hunk 3]
```diff
@@ -1,19 +1,244 @@
...
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintHorizontal_bias="0.5"
+            app:layout_constraintEnd_toEndOf="parent"
+            app:layout_constraintTop_toTopOf="parent"
+            app:layout_constraintBottom_toBottomOf="parent">
+        <LinearLayout
+                android:layout_gravity="center"
+                android:orientation="vertical"
+                android:layout_width="match_parent"
+                android:layout_height="222dp"
```
Comment (SartiMau) [inline app/src/main/res/layout/activity_main.xml]: no hardcode please, use dimens

File: app/src/main/res/layout/activity_main.xml [hunk 4]
```diff
@@ -1,19 +1,244 @@
...
+                android:layout_weight="0.5"
+                android:orientation="vertical"
+                android:layout_gravity="center"
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                app:layout_constraintTop_toTopOf="parent"
+                app:layout_constraintStart_toStartOf="parent"
+                app:layout_constraintEnd_toEndOf="parent"
+                app:layout_constraintBottom_toBottomOf="parent"
+
```
Comment (SartiMau) [inline app/src/main/res/layout/activity_main.xml]: please remove this unnecessary line

File: app/src/main/res/layout/activity_main.xml (hunk 3 above)
Comment (priettt) [inline app/src/main/res/layout/activity_main.xml]: Please, do not hardcode the dimensions, create a dimens class and save them there.

File: app/src/main/res/layout/activity_main.xml [hunk 5]
```diff
@@ -1,19 +1,244 @@
...
+            />
+            <Button
+                    android:text="@string/button_2"
+                    android:layout_width="wrap_content"
+                    android:layout_height="wrap_content"
+                    android:id="@+id/button_2"
+                    android:backgroundTint="@color/numbers"
+                    android:layout_row="@integer/row_1"
+                    android:layout_column="@integer/column_1"
+            />
```
Comment (SartiMau) [inline app/src/main/res/layout/activity_main.xml]: put this un the last line, closing the button definition. In this case 
android:layout_column="@integer/column_1" />

File: app/src/main/res/layout/activity_main.xml [hunk 6]
```diff
@@ -1,19 +1,244 @@
...
+            <Button
+                    android:text="@string/button_0"
+                    android:layout_width="wrap_content"
+                    android:layout_height="wrap_content"
+                    android:id="@+id/button_0"
+                    android:backgroundTint="@color/numbers"
+                    android:layout_row="@integer/row_0"
+                    android:layout_column="@integer/column_1"
+            />
+
```
Comment (SartiMau) [inline app/src/main/res/layout/activity_main.xml:73]: remove unnecessary lines pls

**Review by SartiMau — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/nicocalculator/MainActivity.kt (hunk 1 above)
Comment (priettt) [inline app/src/main/java/com/example/nicocalculator/MainActivity.kt]: Please, move the hardcoded strings to a constant in the class.

File: app/src/main/res/layout/activity_main.xml [hunk 7]
```diff
@@ -1,19 +1,244 @@
...
+
```
Comment (priettt) [inline app/src/main/res/layout/activity_main.xml:235]: Unnecessary consecutive spaces. Use Android Studio's reformat code to fix them, along with the single lines with "/>"

File: app/src/main/java/com/example/nicocalculator/MainActivity.kt [hunk 8]
```diff
@@ -2,11 +2,29 @@ package com.example.nicocalculator
 
 import android.support.v7.app.AppCompatActivity
 import android.os.Bundle
+import kotlinx.android.synthetic.main.activity_main.*
 
 class MainActivity : AppCompatActivity() {
 
+    private val setInitValueOperations = "Operations"
```
Comment (priettt) [inline app/src/main/java/com/example/nicocalculator/MainActivity.kt]: Please, declare kotlin constants as a "const val", either as a public constant in the beggining of the class, or inside of a private companion object. Also, try to follow naming conventions for constants and give a more discriptive name, for example: OPERATIONS_DEFAULT_TEXT

File: app/src/main/java/com/example/nicocalculator/MainActivity.kt (hunk 1 above)
Comment (nicolbidone) [reply app/src/main/java/com/example/nicocalculator/MainActivity.kt]: Done

File: app/src/main/res/layout/activity_main.xml (hunk 3 above)
Comment (nicolbidone) [reply app/src/main/res/layout/activity_main.xml]: Done

File: app/src/main/java/com/example/nicocalculator/MainActivity.kt (hunk 2 above)
Comment (nicolbidone) [reply app/src/main/java/com/example/nicocalculator/MainActivity.kt]: Done

File: app/src/main/res/layout/activity_main.xml (hunk 3 above)
Comment (nicolbidone) [reply app/src/main/res/layout/activity_main.xml]: Done

File: app/src/main/res/layout/activity_main.xml (hunk 4 above)
Comment (nicolbidone) [reply app/src/main/res/layout/activity_main.xml]: Done

File: app/src/main/res/layout/activity_main.xml (hunk 5 above)
Comment (nicolbidone) [reply app/src/main/res/layout/activity_main.xml]: Done

File: app/src/main/res/layout/activity_main.xml (hunk 6 above)
Comment (nicolbidone) [reply app/src/main/res/layout/activity_main.xml:73]: Done

File: app/src/main/java/com/example/nicocalculator/MainActivity.kt (hunk 1 above)
Comment (nicolbidone) [reply app/src/main/java/com/example/nicocalculator/MainActivity.kt]: Done

File: app/src/main/res/layout/activity_main.xml [hunk 9]
```diff
@@ -1,19 +1,237 @@
...
+            <TextView
+                android:id="@+id/text_operations"
+                android:layout_width="match_parent"
+                android:layout_height="wrap_content"
+                android:layout_marginLeft="@dimen/dimen_small_margin"
+                android:layout_weight="@integer/weight_v_pos_1_2" />
+        </LinearLayout>
+
+        <GridLayout
+            android:id="@+id/gridLayout"
```
Comment (priettt) [inline app/src/main/res/layout/activity_main.xml]: Try to name the xml ids with the same naming convention in all the project.  In this case, grid_layout

File: app/src/main/res/layout/activity_main.xml (hunk 7 above)
Comment (nicolbidone) [reply app/src/main/res/layout/activity_main.xml:235]: Done

File: app/src/main/java/com/example/nicocalculator/MainActivity.kt (hunk 8 above)
Comment (nicolbidone) [reply app/src/main/java/com/example/nicocalculator/MainActivity.kt]: Done

File: app/src/main/res/layout/activity_main.xml (hunk 9 above)
Comment (nicolbidone) [reply app/src/main/res/layout/activity_main.xml]: Done

**Review by SartiMau — APPROVED** [review]

**Review by priettt — APPROVED** [review]

File: app/src/main/java/com/example/nicocalculator/MainActivity.kt (hunk 8 above)
Comment (nicolbidone) [reply app/src/main/java/com/example/nicocalculator/MainActivity.kt]: Done

File: app/src/main/res/layout/activity_main.xml (hunk 9 above)
Comment (nicolbidone) [reply app/src/main/res/layout/activity_main.xml]: Done

**Pull request merged**